pex/vendor/__main__.py

```diff
@@ -153,15 +172,22 @@ class VendorizeError(Exception):
 
 def vendorize(root_dir, vendor_specs, prefix):
   for vendor_spec in vendor_specs:
-    cmd = ['pip', 'install', '--upgrade', '--no-compile', '--target', vendor_spec.target_dir,
+    script_dev_null = tempfile.mkdtemp()
+    cmd = ['pip',
+           'install',
+           '--upgrade',
+           '--no-compile',
+           '--target', vendor_spec.target_dir,
+           '--install-option', '--install-scripts={}'.format(script_dev_null),
            vendor_spec.requirement]
     result = subprocess.call(cmd)
+    shutil.rmtree(script_dev_null)
     if result != 0:
       raise VendorizeError('Failed to vendor {!r}'.format(vendor_spec))
 
-    # We know we can get these as a by-product of a pip install but never need them.
-    safe_rmtree(os.path.join(vendor_spec.target_dir, 'bin'))
+    # We know we can get this as a by-product of a pip install but never need it.
```

Comment:
NB: Instead of deleting `bin/` post-install, we now instruct install to use a tmp dir for bin now. This is because the scripts wheel writes to bin/ form part of the checked-in `RECORD` file digests and their contents can change (namely the `#!/path/to/python shebang`). Using a tempdir - a dir outside the `--target` install root - eliminates the `bin/` scripts from being enteed into the `RECORD`.

Reply:
Upon further review had to go a different route to deal with this robustly; thus: 7a72d65